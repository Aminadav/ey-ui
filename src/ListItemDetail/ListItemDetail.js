import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';

const ListItemDetail = props => {
  const { classes, label, value } = props;
  return (
    <ListItem>
      <ListItemText
        data-testid="label-list-text"
        className={classes.item}
        primaryTypographyProps={{ className: classes.label }}
        primary={label}
      />
      <ListItemText data-testid="value-list-text" primary={value} />
    </ListItem>
  );
};

ListItemDetail.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string, PropTypes.instanceOf(Date)]),
  classes: PropTypes.shape({
    label: PropTypes.string.isRequired,
    item: PropTypes.string.isRequired,
  }).isRequired,
};

ListItemDetail.defaultProps = {
  value: '',
};

const styles = theme => ({
  label: {
    fontWeight: 'bold',
    color: theme.palette.primary.light,
  },
  item: {
    flex: 'none',
    minWidth: '11rem',
  },
});

export default withStyles(styles)(ListItemDetail);
