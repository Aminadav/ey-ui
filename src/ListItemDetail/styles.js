import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(() => ({
  label: {
    fontWeight: 'bold',
  },
  item: {
    flex: 'none',
    minWidth: '11rem',
  },
}));

export default useStyles;
