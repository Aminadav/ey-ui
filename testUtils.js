import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import createMuiTheme from '@material-ui/core/styles/createMuiTheme';
import ThemeProvider from '@material-ui/styles/ThemeProvider';

/**
 * Render dom wrapped with material-ui's ThemeProvider.
 */
export function renderWithTheme(ui) {
  return {
    ...render(<ThemeProvider theme={createMuiTheme()}>{ui}</ThemeProvider>),
  };
}

/**
 * Render dom wrapped with multiple dependent context providers.
 */
export function renderWithProviders(ui) {
  return {
    ...render(
      <MemoryRouter initialEntries={['/']}>
        <ThemeProvider theme={createMuiTheme()}>{ui}</ThemeProvider>
      </MemoryRouter>,
    ),
  };
}

export default { renderWithTheme };
