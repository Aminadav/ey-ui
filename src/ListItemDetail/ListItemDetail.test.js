import React from 'react';
import { renderWithTheme } from '../../testUtils';
import ListItemDetail from './ListItemDetail';

describe('ListItemDetail', () => {
  it('component matches snapshot with a label prop', () => {
    const { container } = renderWithTheme(<ListItemDetail label="TestLabel" value="TestValue" />);
    expect(container).toMatchSnapshot();
  });

  it('should render properly with a label and value element', () => {
    const { getByTestId } = renderWithTheme(<ListItemDetail label="TestLabel" value="TestValue" />);
    expect(getByTestId('label-list-text')).toBeInTheDocument();
    expect(getByTestId('value-list-text')).toBeInTheDocument();
  });

  it('should render with proper label and value content', () => {
    const { getByTestId } = renderWithTheme(<ListItemDetail label="TestLabel" value="TestValue" />);
    expect(getByTestId('label-list-text').textContent).toBe('TestLabel');
    expect(getByTestId('value-list-text').textContent).toBe('TestValue');
  });
});
