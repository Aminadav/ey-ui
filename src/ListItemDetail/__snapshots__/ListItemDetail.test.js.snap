// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemDetail component matches snapshot with a label prop 1`] = `
<div>
  <li
    class="MuiListItem-root MuiListItem-gutters"
  >
    <div
      class="MuiListItemText-root ListItemDetail-item-2"
      data-testid="label-list-text"
    >
      <span
        class="MuiTypography-root ListItemDetail-label-1 MuiTypography-body1"
      >
        TestLabel
      </span>
    </div>
    <div
      class="MuiListItemText-root"
      data-testid="value-list-text"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        TestValue
      </span>
    </div>
  </li>
</div>
`;
